rest_framework/fields.py

```diff
@@ -385,8 +385,10 @@ def get_value(self, dictionary):
                 # If the field is blank, and null is a valid value then
                 # determine if we should use null instead.
                 return '' if getattr(self, 'allow_blank', False) else None
-            elif ret == '' and self.default:
-                return empty
+            elif ret == '' and not self.required:
```

Comment:
I'll link to the tests around CharField behaviors, yup. Lemme wait til we have a tagged version to point at.